stackslib/src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -884,7 +885,8 @@ impl LeaderBlockCommitOp {
 
         let is_already_committed = tx.expects_stacks_block_in_fork(&self.block_header_hash)?;
 
-        if is_already_committed {
+        // in Epoch3.0+, block commits can include Stacks blocks already accepted in the fork.
```

Comment:
I'm not sure how this is possible?  All this is checking is that there are no duplicate block-commit block hashes (i.e. StacksBlockIds) in this Bitcoin fork.

Reply:
There's at least two cases in which this would be possible:

1. A miner wins sortition, but doesn't produce any blocks.
2. During the transition from 2.5 to 3.0, the first block will fail this test.

Reply:
Gotcha